client/app/scripts/components/search.js

```diff
@@ -121,13 +125,14 @@ class Search extends React.Component {
           <div className="search-input">
             {showPinnedSearches && pinnedSearches.toIndexedSeq()
               .map(query => <SearchItem query={query} key={query} />)}
-            <input className="search-input-field" type="text" id={inputId}
+            <input
+              className="search-input-field" type="text" id={inputId}
               value={value} onChange={this.handleChange}
               onFocus={this.handleFocus} onBlur={this.handleBlur}
-              disabled={disabled} ref="queryInput" />
+              disabled={disabled} ref={this.saveQueryInputRef} />
```

Comment:
I think both patterns existed in parallel for a while now (the [official documentation](https://facebook.github.io/react/docs/refs-and-the-dom.html) uses the functional one).

The new `no-string-refs` rule goes against our *old* pattern, here's a discussion why:
http://stackoverflow.com/questions/37468913/why-ref-string-is-legacy